lib/spack/spack/solver/asp.py

```diff
@@ -4028,6 +4043,12 @@ def __init__(self, configuration: spack.config.Configuration) -> None:
                     SpecFilter.from_buildcache(
                         configuration=self.configuration, include=[], exclude=[]
                     ),
+                    SpecFilter.from_environment(
+                        configuration=self.configuration,
+                        include=[],
+                        exclude=[],
+                        env=ev.active_environment(),
```

Comment:
```suggestion
                        env=ev.active_environment(),  # includes all concrete includes
```
I think that will help future readers